### PR TITLE
feat(operation-id-case-convention): add new operation-id-case-convention rule

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit introduces the new spectral-style operation-id-case-convention
rule, which replaces the legacy operation_id_case_convention rule.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

#### Checklist for removing an old validation rule:
- [x] Removed old rule implementation (packages/validator/src/plugins/validation/*.js)
- [x] Removed or adjusted testcases (packages/validator/test)
- [x] Updated default configuration to deprecate old rule (packages/validator/src/.defaultsForValidator.js)
- [x] Removed docs of old rule (docs/ibm-legacy-rules.md)

### CODE REVIEW

File: packages/ruleset/src/functions/operation-id-case-convention.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  const errors = [];
+
+  const result = casing(operationId, casingConfig);
+  if (result) {
+    // Update the message to clarify that it is an operationId value that is incorrect,
+    // and update the path.
+    result[0].message = 'Operation ids ' + result[0].message;
+    result[0].path = [...path, 'operationId'];
+    errors.push(result[0]);
```
Comment (dpopp07) [inline packages/ruleset/src/functions/operation-id-case-convention.js:31]: Not important but I thought I had for these operation rules that don't loop through anything: we could return a single value when the rule is broken and avoid defining the array, like
```suggestion
    return [result[0]];
```

and at the end of the function, `return [];`.

I really don't know how much of a performance implication there actually is for allocating the array every time so feel free to ignore this, just a suggestion.

**Review by dpopp07 — APPROVED** [review]
Looks great! 👍

File: packages/ruleset/src/functions/operation-id-case-convention.js (hunk 1 above)
Comment (padamstx) [reply packages/ruleset/src/functions/operation-id-case-convention.js:31]: fixed that

**Pull request merged**